### DISCUSSION TITLE
About useless dependencies

### DISCUSSION THREAD

**Discussion by ghost** [termux/termux-packages#19295] [Ideas]
During Apt Install, Apt installed too many dependent files. In fact, there are many dependencies that are completely useless, but they bring serious storage space waste.
For example, I build Qemu-System-x86-64-headless, it is only about 190MB. But when I use apt install, it is 397MB, it is too big and unnecessary. It wasted about 200MB of storage space.
I don't like unnecessary storage space waste.
I used to improve this issue through the following two lines of configuration files a years ago:
```
APT::Install-Suggests "0";
APT::Install-Recommends "0";
```
But now it has no effect.

**Comment by Biswa96** [comment]
> it is too big and unnecessary.

Could you rectify which packages or dependencies are unnecessary?

**Reply by ghost** [reply]
A example on qemu-system-x86-64-headless: 
dbus pulseaudio libwebrtc-audio-processing libx11 libflac (and more.....)

**Reply by ghost** [reply]
libsndfile opengl

**Reply by ghost** [reply]
I only used these few dependencies to compile successfully:
```
python python-pip python-setuptools ninja gcc bzip2 pkgconf spice-protocol pixman spice make
```
And the gcc, ninja, make, and Python are no longer useful after compilation is completed.

**Reply by licy183** [reply]
Suggested packages will not be installed automatically, you can also not install the recommended packages through `apt install --no-install-recommends qemu-system-x86-64-headless`.

I'm afraid that these packages are **not unnecessary**.

`qemu` depends on `libspice-server`. `libspice-server` depends on `gstreamer`. `gstreamer` recommends on `opengl`. `opengl` depends on `libglvnd` and `mesa`. `mesa` depends on `libx11`.

`qemu` also depends on `pulseaudio`. `pulseaudio` depends on `dbus`, `libwebrtc-audio-processing` and `libsndfile`. `libsndfile` depends on `libflac`.

It is indeed difficult for package maintainers to decide whether to enable/disable a certain feature for one software. In most cases, they will turn on most of the commonly-used features for packages in the distro repository to ensure that it meets the needs of most users. If you really care about the wasted space, the best solution is to compile and package it yourself.

**Reply by ghost** [reply]
But Termux does not have gcc, only clang. When compiling QEMU, gcc must be used , not clang.
```
# u0_a178 @ localhost in ~ [19:03:08]
$ apt install gcc
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done                     Package gcc is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:              libllvm

E: Package 'gcc' has no installation candidate

# u0_a178 @ localhost in ~ [19:03:18] C:100           $ gcc
The program gcc is not installed. Install it by executing:
 pkg install clang
```

**Comment by ghost** [comment]
Thanks for you all. I found gcc in tur-repo. Now I can manually compile QEMU.
